compiler/compile/OSRData.hpp

```diff
@@ -44,6 +44,21 @@ namespace TR
       osr,
       traditional
       };
+  enum OSRTransitionTarget
```

Comment:
Can you please add some descriptive comments on these enums describing what they are and what the different values means since these are now very important in understanding the OSR infrastructure